l10n_th_personal_income_tax/wizard/account_payment_register.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2021 Ecosoft Co., Ltd (https://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+from ...l10n_th_withholding_tax_cert.models.withholding_tax_cert import (
+    WHT_CERT_INCOME_TYPE,
+)
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = "account.payment.register"
+
+    def _create_payments(self):
+        payments = super()._create_payments()
+        for payment in payments:
+            move_ids = payment.reconciled_bill_ids.filtered("account_pit")
+            if move_ids:
+                payment.write(
+                    {
+                        "pit_line": [
+                            (
+                                0,
+                                0,
+                                {
+                                    "partner_id": payment.partner_id.id,
+                                    "wht_cert_income_type": move_ids[
+                                        0
+                                    ].pit_wht_cert_income_type
+                                    or False,
+                                    "amount_income": sum(
+                                        move_ids.mapped("amount_untaxed")
+                                    )
+                                    or 0.0,
+                                    "amount_wht": self.payment_difference,
```

Comment:
Not sure, but should this be,
```suggestion
                                    "amount_wht": payment.payment_difference,
```

Reply:
Edited.

Reply:
field `payment_difference` on payment register only.